### PR TITLE
feat: add missing methods from the latest RoleManager interface

### PR DESCRIPTION
**Pull request by checkaayush**
## Changes

- Converts the repo into a go module
- Partial upgrade to github.com/casbin/casbin/v2
- Add missing methods to comply with latest RoleManager interface

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module github.com/casbin/okta-role-manager
+
+go 1.17
+
+require (
+	github.com/casbin/casbin v1.9.1
```
Comment (hsluoyz) [inline go.mod:6]: Remove casbin v1.9.1

File: go.sum [hunk 2]
```diff
@@ -0,0 +1,21 @@
+github.com/Knetic/govaluate v3.0.1-0.20171022003610-9aa49832a739+incompatible h1:1G1pk05UrOh0NlF1oeaaix1x8XzrfjIDK47TY0Zehcw=
+github.com/Knetic/govaluate v3.0.1-0.20171022003610-9aa49832a739+incompatible/go.mod h1:r7JcOSlj0wfOMncg0iLm8Leh48TZaKVeNIfJntJ2wa0=
+github.com/casbin/casbin v1.9.1 h1:ucjbS5zTrmSLtH4XogqOG920Poe6QatdXtz1FEbApeM=
```
Comment (hsluoyz) [inline go.sum:3]: Remove casbin v1.9.1

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: go.sum (hunk 2 above)
Comment (checkaayush) [reply go.sum]: Done.

File: go.mod (hunk 1 above)
Comment (checkaayush) [reply go.mod]: Done.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**